packages/gestalt/src/TextField.js

```diff
@@ -117,18 +122,57 @@ const TextFieldWithForwardRef: React$AbstractComponent<Props, HTMLInputElement>
     placeholder,
     size = 'md',
     tags,
-    type = 'text',
+    type: typeProp = 'text',
     value,
   }: Props,
   ref,
 ): Node {
+  const [type, setType] = useState<Type>(typeProp);
```

Comment:
Yes, this is initializing a state variable with a prop value and then disregarding the prop value — often a code smell, I know. This is necessary to internalize the effective input type (password vs text) and not force the user to handle responding to clicks on the button.

Reply:
This makes sense!